test/tests/mesh/checkpoint/tests

```diff
@@ -10,18 +10,22 @@
     cli_args = 'Outputs/file_base=test_2 --use-split --split-file checkpoint_split_in'
     max_parallel = 2
     min_parallel = 2
+    # checkpoint files are platform-dependent
```

Comment:
Removing _every_ test for mac here is dangerous. do we not still support ascii checkpointing?

Reply:
Before this PR we only supported binary (cpr) checkpoints. Now we only support compressed ascii (cpa.gz) checkpoints.

The reason for adding the platform specification here is that this test has to read mesh files that were generated on a linux platform and stored as part of the moose repo - they are not generated in a prerequisite test.

See below conversation with @roystgnr that prompted me to add patform dependence:

>Me: When we default to using ASCII instead of binary to store checkpoints, does this give us the benefit that a checkpoint written on one architecture/OS can be read and used on a different architecture/OS? Or do we expect there to be differences in the compression/decompression on different architectures/OSs that cause issues?

> Roy: Both ASCII and binary can be architecture independent.  The whole point of XDR is to handle things like endianness differences.  But in the specific case of .cpa files... on the one hand, you'd get a little extra compatibility, because IIRC we have a bunch of integer-size stuff in .xdr headers that isn't in .cpr, and on the other hand no, because half the point of CheckpointIO was to have a non-format where we don't need 10 layers of compatibility, and if you try to reload one after so much as a git pull since saving it our only guarantee is that we'll laugh at you.

That being say, an easy way to have this tests on all platforms would to be to add a test that generates the split mesh instead of storing the mesh files as part of the repo.

Reply:
I just dug into checkpoint_io.C, and it looks like I actually *did* get around to putting the integer-size stuff into .cpr ... or rather, Derek forked CheckpointIO off of XDRIO after I added it to .xdr?

Anyway, we've probably actually maintained architecture independence since then; now that I think about it, we added some checked-in .cpr files to the MOOSE test suite back when I was working on a feature for it ... possibly N-to-M restarts?

Reply:
Yeah, there seemed to be architecture independence for this particular test while the files were cpr, but as cpa.gz generated on linux, I get the following error message on mac:

`mesh/checkpoint.test_2: *** ERROR ***
mesh/checkpoint.test_2: While reading restartable data in "/Users/behnpa/projects/moose_checkpoint/test/tests/mesh/checkpoint/checkpoint_split_in.cpa.gz/meta_data_mesh.rd":
mesh/checkpoint.test_2: 
mesh/checkpoint.test_2: The backup is not compatible
mesh/checkpoint.test_2: 
mesh/checkpoint.test_2: The hash code check for a basic type (int) failed.
mesh/checkpoint.test_2: It is likely that this backup was stored with a different architecture or operating system`

Reply:
...

If anything .cpa.gz should be more independent.

That error message isn't coming from CheckpointIO, though.  The .cpa.gz files are being written/read to actual files, right?  Not interleaved into some stream where a bad read could move on to non-CheckpointIO data too early or too late?

Reply:
These are the files I'm seeing:

`(moose) INL654724:checkpoint behnpa$ ls -R checkpoint_split_in.cpa.gz
2			4			8			meta_data_mesh.rd

checkpoint_split_in.cpa.gz/2:
header.gz	split-2-0.gz	split-2-1.gz

checkpoint_split_in.cpa.gz/4:
header.gz	split-4-0.gz	split-4-1.gz	split-4-2.gz	split-4-3.gz

checkpoint_split_in.cpa.gz/8:
header.gz	split-8-1.gz	split-8-3.gz	split-8-5.gz	split-8-7.gz
split-8-0.gz	split-8-2.gz	split-8-4.gz	split-8-6.gz

checkpoint_split_in.cpa.gz/meta_data_mesh.rd:
data	header`

In my experience, when `ls` is run on a compressed file, the command can't see into the archive. Why is that not the case here?

Reply:
I just recompiled without the hash code checks, and all the exodiff tests passed. I'll look into the check a bit more and get with Roy if I can't figure it out.

Reply:
The check seems simple enough at the surface, I'm handing the investigation off to @loganharbour as to whether this check is needed.

Reply:
The hash code checks exist as a sanity check to the user to make sure that they're not running a binary checkpoint from another machine. They run typeid(T) on the type being stored and make sure it's the same on the other end.

Reply:
Here is the resolution that I will implement that we discussed in the NEAMS meeting:

Instead of storing the split mesh files on github, these will be generated on-the-fly in prerequisite tests.